src/txdb.cpp

```diff
@@ -116,7 +116,9 @@ bool CCoinsViewDB::BatchWrite(CCoinsMap &mapCoins, const uint256 &hashBlock) {
         }
         count++;
         CCoinsMap::iterator itOld = it++;
-        mapCoins.erase(itOld);
+        if (erase) {
```

Comment:
nit: I think it'd be clearer to avoid this use-once local variable by using the same pattern that you used in coins.cpp:

```
it = erase ? mapCoins.erase(it) : ++it;
```

Reply:
Does this need an exception, to always delete `FRESH` spent coins? Or those can't exist at this level?

Reply:
Good suggestion, thanks! Fixed.

`FRESH` spent coins won't exist here because they will have been removed from the child cache first - for proof of this note that nothing calls `CCoinsViewDB::BatchWrite()` aside from CCoinsViewCache, and `SpendCoin()` only exists on CCoinsViewCache instances.